packages/elements/src/create-custom-element.ts

```diff
@@ -134,114 +100,143 @@ export function createCustomElement<P>(
 
   const attributeToPropertyInputs = getDefaultAttributeToPropertyInputs(inputs);
 
-  class NgElementImpl extends NgElement {
-    // Work around a bug in closure typed optimizations(b/79557487) where it is not honoring static
-    // field externs. So using quoted access to explicitly prevent renaming.
-    static readonly['observedAttributes'] = Object.keys(attributeToPropertyInputs);
-
-    protected get ngElementStrategy(): NgElementStrategy {
-      // NOTE:
-      // Some polyfills (e.g. `document-register-element`) do not call the constructor, therefore
-      // it is not safe to set `ngElementStrategy` in the constructor and assume it will be
-      // available inside the methods.
-      //
-      // TODO(andrewseguin): Add e2e tests that cover cases where the constructor isn't called. For
-      // now this is tested using a Google internal test suite.
-      if (!this._ngElementStrategy) {
-        const strategy = this._ngElementStrategy =
-            strategyFactory.create(this.injector || config.injector);
-
-        // Re-apply pre-existing input values (set as properties on the element) through the
-        // strategy.
-        inputs.forEach(({propName}) => {
-          if (!this.hasOwnProperty(propName)) {
-            // No pre-existing value for `propName`.
-            return;
-          }
-
-          // Delete the property from the instance and re-apply it through the strategy.
-          const value = (this as any)[propName];
-          delete (this as any)[propName];
-          strategy.setInputValue(propName, value);
-        });
+  const NgElementImplCtor = newCustomElementWithBehavior(baseClass => {
+    class NgElementImpl extends baseClass implements NgElementBehavior<NgElementImpl> {
+      // Work around a bug in closure typed optimizations(b/79557487) where it is not honoring
+      // static field externs. So using quoted access to explicitly prevent renaming.
+      static readonly['observedAttributes'] = Object.keys(attributeToPropertyInputs);
+
+      private ngElementEventsSubscription: Subscription|null = null;
```

Comment:
nit: carrying over comments from the previous implementation:
```suggestion
      /**
       * A subscription to change, connect, and disconnect events in the custom element.
       */
      private ngElementEventsSubscription: Subscription|null = null;
```